### PR TITLE
Add processor count to device info

### LINKED ISSUE #1401: Use `getExtraContext` on iOS
### Description

`sentry-cocoa` has a new method for HybridSDKs which exposes extra context data that originally were available only in `sentry-cocoa` client. 

https://github.com/getsentry/sentry-cocoa/pull/2855

RN implementation: https://github.com/getsentry/sentry-react-native/pull/2984

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
https://github.com/getsentry/team-mobile/issues/77
Fixes https://github.com/getsentry/sentry-dart/issues/1401

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

**Comment by ueman** [comment]
https://github.com/getsentry/sentry-dart/blob/fed8146b5b1d465a10ce707039dbe2f9b2ceb9c8/dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart#L104
Does that need to be changed too?

**Comment by marandaneto** [comment]
> https://github.com/getsentry/sentry-dart/blob/fed8146b5b1d465a10ce707039dbe2f9b2ceb9c8/dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart#L104
> 
> 
> Does that need to be changed too?

Nope because `device?.processorCount` is going to be populated, also because `hasNativeIntegration` is true so it won't fallback to `_getDevice`, its still gonna be useful for non native plats.

File: flutter/ios/Classes/SentryFlutterPluginApple.swift [hunk 1]
```diff
@@ -687,4 +718,12 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
     }
 }
 
+extension Dictionary {
+    mutating func mergeDictionary(other: Dictionary) {
+        for (key, value) in other {
+            self.updateValue(value, forKey: key)
+        }
+    }
+}
```
Comment (krystofwoldrich) [inline flutter/ios/Classes/SentryFlutterPluginApple.swift]: L: Looks good, could we use [`dict.merging`](https://developer.apple.com/documentation/swift/dictionary/merge(_:uniquingkeyswith:)-m2ub) from Swift 4, or the project uses older version?

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀

File: flutter/ios/Classes/SentryFlutterPluginApple.swift (hunk 1 above)
Comment (marandaneto) [reply flutter/ios/Classes/SentryFlutterPluginApple.swift]: Good point [`a29ac2a` (#1402)](https://github.com/getsentry/sentry-dart/pull/1402/commits/a29ac2aa2583676b4e6daea80fdd1ff1d37451d8)

**Pull request merged**